### PR TITLE
LibGL: Implement `GL_TEXTURE` matrix mode

### PR DESCRIPTION
**Pull request by GMTA**
This PR implements the `GL_TEXTURE` matrix mode and texture coordinates matrix transformation. In addition, `GL_POLYGON` is now supported and the maximum matrix stack depths can be queried.

Grim Fandango in ScummVM is now able to render the 3D actors!
image

(Note that you need to compile and enable OpenGL mode in ScummVM to try this for yourself)

### CODE REVIEW

File: Userland/Libraries/LibGL/SoftwareGLContext.h [hunk 1]
```diff
@@ -161,10 +161,11 @@ class SoftwareGLContext : public GLContext {
     GLenum m_current_matrix_mode;
     FloatMatrix4x4 m_projection_matrix = FloatMatrix4x4::identity();
     FloatMatrix4x4 m_model_view_matrix = FloatMatrix4x4::identity();
-    FloatMatrix4x4 m_current_matrix = FloatMatrix4x4::identity();
+    FloatMatrix4x4 m_texture_matrix = FloatMatrix4x4::identity();
 
     Vector<FloatMatrix4x4> m_projection_matrix_stack;
     Vector<FloatMatrix4x4> m_model_view_matrix_stack;
+    Vector<FloatMatrix4x4> m_texture_matrix_stack;
```
Comment (sunverwerth) [inline Userland/Libraries/LibGL/SoftwareGLContext.h:168]: The texture matrix stack is actually part of the texture units and each unit has its own stack. `glActiveTexture` selects which is currently in use. https://www.khronos.org/registry/OpenGL/specs/gl/glspec15.pdf Page 43-44

**Review by sunverwerth — CHANGES_REQUESTED** [review]
Looks good overall. Please see my note about the texture matrix stack and texture units.

File: Userland/Libraries/LibGL/SoftwareGLContext.h (hunk 1 above)
Comment (sunverwerth) [reply Userland/Libraries/LibGL/SoftwareGLContext.h:168]: I'd say since we don't support multi texturing yet, adding a FIXME here would suffice.

Comment (GMTA) [reply Userland/Libraries/LibGL/SoftwareGLContext.h]: Agreed - I've added two `FIXME`s

**Review by sunverwerth — APPROVED** [review]

**Pull request merged**